Plugin build (master branch) hangs on MaxOsx 11.4

cargo check

```
    Checking serde_json v1.0.86
    Checking ahash v0.8.0
    Checking backtrace v0.3.66
    Checking miette v4.7.1
    Checking sourcemap v6.2.0
    Checking rkyv v0.7.37
    Checking swc_atoms v0.4.23
    Checking swc_common v0.27.16
    Checking swc_ecma_ast v0.90.20
    Checking swc_error_reporters v0.11.7
    Checking testing v0.29.7
    Building [=======================> ] 329/340: swc_ecma_ast 
```

and hangs forever

I tried to update packages:

```
22:15:02 [~] [main] $ git diff
diff --git a/Cargo.toml b/Cargo.toml
index ca81c8f..50eac91 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -14,14 +14,20 @@ ahash = { version = "0.8.0", features = ["serde"] }
 phf = { version = "0.11.1", features = ["macros"] }
 serde = { version = "1.0.144", features = ["derive"] }
 serde_json = "1.0.85"
-swc_common = { version = "0.27.12", features = ["concurrent"] }
-swc_core = { version = "0.10.0", features = [
-    "plugin_transform",
-    "utils",
-    "visit",
-    "ast",
+swc_common = { version = "0.29.10", features = ["concurrent"] }
+swc_core = { version = "0.40.*", features = [
+    "common_perf",
+    "allocator_node",
+    "base",
+    "base_node",
     "common",
-    "quote"
+    "ecma_visit",
+    "ecma_transforms",
+    "bundler",
+    "ecma_loader",
+    "ecma_ast",
+    "trace_macro",
+    "plugin_transform",
 ] }
 
 [profile.release]
@@ -31,5 +37,5 @@ opt-level = 3
 panic = "abort"
 
 [dev-dependencies]
-swc_ecmascript = { version = "0.192.0", features = ["parser", "transforms"] }
-testing = "0.29.4"
+swc_ecmascript = { version = "0.205.42", features = ["parser", "transforms"] }
+testing = "0.31.10"`
```

Now, build/check does not hangs, but, compilation breaks on missing include file shed.h

```
    Checking time v0.1.44
    Checking tempfile v3.3.0
    Checking aho-corasick v0.7.19
    Checking object v0.29.0
    Checking nom v7.1.1
    Checking regex-automata v0.1.10
   Compiling num_threads v0.1.6
    Checking lexical-parse-float v0.8.5
    Checking lexical-write-float v0.8.5
The following warnings were emitted during compilation:

warning: In file included from ./mimalloc/src/static.c:17:
warning: In file included from ./mimalloc/include/mimalloc-internal.h:11:
warning: In file included from ./mimalloc/include/mimalloc-types.h:13:
warning: ./mimalloc/include/mimalloc-atomic.h:326:10: fatal error: 'sched.h' file not found
warning: #include <sched.h>
warning:          ^~~~~~~~~
warning: 1 error generated.

error: failed to run custom build command for `mimalloc-rust-sys v1.7.6-source`

Caused by:
  process didn't exit successfully: `/Users/victor.tiurin/Projects/libs/swc-plugin/target/debug/build/mimalloc-rust-sys-58e7f6a893e4af77/build-script-build` (exit status: 1)
  --- stdout
  OPT_LEVEL = Some("0")
  TARGET = Some("wasm32-wasi")
  HOST = Some("x86_64-apple-darwin")
  CC_wasm32-wasi = None
  CC_wasm32_wasi = None
  TARGET_CC = None
  CC = None
  CFLAGS_wasm32-wasi = None
  CFLAGS_wasm32_wasi = None
  TARGET_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("true")
  running: "clang" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "--target=wasm32-wasi" "-I" "./mimalloc/include" "-I" "./mimalloc/src" "-Wall" "-Wextra" "-DMI_BUILD_SHARED=0" "-DMI_DEBUG_FUL=3" "-o" "/Users/victor.tiurin/Projects/libs/swc-plugin/target/wasm32-wasi/debug/build/mimalloc-rust-sys-836729d64c30fcef/out/./mimalloc/src/static.o" "-c" "./mimalloc/src/static.c"
  cargo:warning=In file included from ./mimalloc/src/static.c:17:
  cargo:warning=In file included from ./mimalloc/include/mimalloc-internal.h:11:
  cargo:warning=In file included from ./mimalloc/include/mimalloc-types.h:13:
  cargo:warning=./mimalloc/include/mimalloc-atomic.h:326:10: fatal error: 'sched.h' file not found
  cargo:warning=#include <sched.h>
  cargo:warning=         ^~~~~~~~~
  cargo:warning=1 error generated.
  exit status: 1
```

@sergeysova @ilyalesik 
Anyone can help me with swc-plugin build on MacOsx?

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -12,16 +12,16 @@ crate-type = ["cdylib", "rlib"]
 radix_fmt = "1.0.0"
 ahash = { version = "0.8.0", features = ["serde"] }
 phf = { version = "0.11.1", features = ["macros"] }
-serde = { version = "1.0.144", features = ["derive"] }
+serde = { version = "1.0.145", features = ["derive"] }
 serde_json = "1.0.85"
-swc_common = { version = "0.27.12", features = ["concurrent"] }
-swc_core = { version = "0.10.0", features = [
+swc_common = { version = "0.29.4", features = ["concurrent"] }
+swc_core = { version = "0.28.18", features = [
     "plugin_transform",
-    "utils",
-    "visit",
-    "ast",
+    "ecma_utils",
+    "ecma_visit",
+    "ecma_ast",
     "common",
-    "quote"
+    "ecma_quote"
 ] }
 
 [profile.release]
@@ -31,5 +31,5 @@ opt-level = 3
 panic = "abort"
 
 [dev-dependencies]
-swc_ecmascript = { version = "0.192.0", features = ["parser", "transforms"] }
-testing = "0.29.4"
+swc_ecmascript = { version = "0.203.29", features = ["parser", "transforms"] }
+testing = "0.31.4"

--- a/package.json
+++ b/package.json
@@ -15,12 +15,12 @@
   "author": "Kirill Mironov <vetrokm@gmail.com>",
   "license": "MIT",
   "devDependencies": {
-    "@babel/core": "~7.18.10",
+    "@babel/core": "~7.19.3",
     "@effector/swc-plugin": "link:.",
     "@swc/cli": "~0.1.57",
-    "@swc/core": "~1.2.233",
+    "@swc/core": "~1.3.4",
     "effector": "~22.3.0",
     "unplugin-swc": "~1.3.2",
-    "vite": "~3.0.8"
+    "vite": "~3.1.4"
   }
 }

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@effector/swc-plugin",
-  "version": "0.0.2",
+  "version": "0.0.5",
   "main": "target/wasm32-wasi/release/effector_swc_plugin.wasm",
   "files": [
     "target/wasm32-wasi/release/effector_swc_plugin.wasm"

--- a/src/effector.rs
+++ b/src/effector.rs
@@ -9,7 +9,7 @@ use std::{
 
 use ahash::{AHashMap, AHashSet};
 use swc_common::{sync::Lrc, Loc, SourceMapper, DUMMY_SP};
-use swc_core::{
+use swc_core::ecma::{
     ast::*,
     utils::{private_ident, quote_ident},
     visit::{noop_visit_mut_type, VisitMut, VisitMutWith},
@@ -634,7 +634,7 @@ impl<'a, C: SourceMapper> Effector<'a, C> {
         let local_ident = self.state.uid_generator.factory_generate_identifier(method.as_ref());
         let decl = ImportDecl {
             span: DUMMY_SP,
-            src: Str::from("effector"),
+            src: Box::new(Str::from("effector")),
             type_only: false,
             asserts: None,
             specifiers: vec![ImportSpecifier::Named(ImportNamedSpecifier {
@@ -696,7 +696,7 @@ impl<'a, C: SourceMapper> VisitMut for Effector<'a, C> {
                     .body
                     .iter()
                     .rposition(|m| matches!(m, ModuleItem::ModuleDecl(ModuleDecl::Import(_))));
-                let stmt = ModuleItem::Stmt(Stmt::Decl(Decl::Var(VarDecl {
+                let stmt = ModuleItem::Stmt(Stmt::Decl(Decl::Var(Box::new(VarDecl {
                     span: DUMMY_SP,
                     kind: VarDeclKind::Var,
                     declare: false,
@@ -706,7 +706,7 @@ impl<'a, C: SourceMapper> VisitMut for Effector<'a, C> {
                         init: Some(Box::new(Expr::from(filename.unwrap_or_else(|| "".into())))),
                         definite: false,
                     }],
-                })));
+                }))));
                 if let Some(index) = last_import_index {
                     m.body.insert(index + 1, stmt);
                 } else {
@@ -783,8 +783,9 @@ impl<'a, C: SourceMapper> VisitMut for Effector<'a, C> {
             let mut check_and_replace =
                 |replace: bool, no_scope: &AHashSet<String>, scope: &AHashSet<String>| {
                     if replace && no_scope.contains(source) {
-                        d.src =
-                            scope.iter().find(|s| s.contains("scope")).unwrap().to_string().into()
+                        d.src = Box::new(
+                            scope.iter().find(|s| s.contains("scope")).unwrap().to_string().into(),
+                        )
                     }
                 };
 

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -6,9 +6,8 @@ mod sid;
 
 use swc_common::plugin::metadata::TransformPluginMetadataContextKind;
 use swc_core::{
-    ast::Program,
+    ecma::{ast::Program, visit::VisitMutWith},
     plugin::{plugin_transform, proxies::TransformPluginProgramMetadata},
-    visit::VisitMutWith,
 };
 
 pub use crate::{

--- a/src/macros.rs
+++ b/src/macros.rs
@@ -13,15 +13,15 @@ macro_rules! obj_lit_internal {
     // Insert the current entry followed by trailing comma.
     (@object $object:ident [$($key:tt)+] ($value:expr) , $($rest:tt)*) => {
         $object.props.push(
-            ::swc_core::ast::PropOrSpread::Prop(
-                Box::new(::swc_core::ast::Prop::KeyValue(
-                    ::swc_core::ast::KeyValueProp {
-                        key: ::swc_core::ast::PropName::Ident(
-                            ::swc_core::utils::quote_ident!(($($key)+))
+            ::swc_core::ecma::ast::PropOrSpread::Prop(
+                Box::new(::swc_core::ecma::ast::Prop::KeyValue(
+                    ::swc_core::ecma::ast::KeyValueProp {
+                        key: ::swc_core::ecma::ast::PropName::Ident(
+                            ::swc_core::ecma::utils::quote_ident!(($($key)+))
                         ),
                         value: Box::new(
-                            ::swc_core::ast::Expr::Lit(
-                                ::swc_core::ast::Lit::Str(
+                            ::swc_core::ecma::ast::Expr::Lit(
+                                ::swc_core::ecma::ast::Lit::Str(
                                     $value
                                 )
                             )
@@ -37,15 +37,15 @@ macro_rules! obj_lit_internal {
     // Insert the last entry without trailing comma.
     (@object $object:ident [$($key:tt)+] ($value:expr)) => {
         $object.props.push(
-            ::swc_core::ast::PropOrSpread::Prop(
-                Box::new(::swc_core::ast::Prop::KeyValue(
-                    ::swc_core::ast::KeyValueProp {
-                        key: ::swc_core::ast::PropName::Ident(
-                            ::swc_core::utils::quote_ident!(($($key)+))
+            ::swc_core::ecma::ast::PropOrSpread::Prop(
+                Box::new(::swc_core::ecma::ast::Prop::KeyValue(
+                    ::swc_core::ecma::ast::KeyValueProp {
+                        key: ::swc_core::ecma::ast::PropName::Ident(
+                            ::swc_core::ecma::utils::quote_ident!(($($key)+))
                         ),
                         value: Box::new(
-                            ::swc_core::ast::Expr::Lit(
-                                ::swc_core::ast::Lit::Str(
+                            ::swc_core::ecma::ast::Expr::Lit(
+                                ::swc_core::ecma::ast::Lit::Str(
                                     $value
                                 )
                             )
@@ -96,7 +96,7 @@ macro_rules! obj_lit_internal {
 
     ({ $($tt:tt)+ }) => {
         {
-            let mut object = (::swc_core::ast::ObjectLit {
+            let mut object = (::swc_core::ecma::ast::ObjectLit {
                 span: ::swc_common::DUMMY_SP,
                 props: vec![]
             });
@@ -108,6 +108,6 @@ macro_rules! obj_lit_internal {
     };
 
     ($other:expr) => {
-        ::swc_core::ast::Str::from($other)
+        ::swc_core::ecma::ast::Str::from($other)
     };
 }

